Assets/Scripts/ShadowPuzzle/PilarCheck.cs

```diff
@@ -0,0 +1,32 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor.Experimental.GraphView;
+using UnityEngine;
+
+public class PilarCheck : MonoBehaviour
+{
+    public bool pilarInPlace;
+    public GameObject pilarPlace;
+    public GameObject pilar;
```

Comment:
```suggestion
    public bool Completed { get; private set; }
    
    [SerializeField] private Transform lockPosition;
    [SerializeField] private GameObject pillar;
    
    [SerializeField] private UnityEvent onCompleted = new();
```
Paar dingen:

- Betere benamingen
- Denk aan access modifiers. niet alles moet public
- Completed moet ook private set sinds je niet wilt dat het van buiten aanpasbaar is.
- UnityEvent gebruiken om script los te koppelen